### ISSUE TITLE
After update doctrine/doctrine-bundle to version 1.11.x, migrations not working properly with "schema_filter" option

### ISSUE THREAD

**Issue by mleko64** [doctrine/migrations#821]
### Bug Report

Symfony version: 4.2.8

doctrine/doctrine-bundle: 1.11.0
doctrine/dbal: 2.9.2
doctrine/doctrine-migrations-bundle: 2.0.0

#### Summary

After update "doctrine/doctrine-bundle" from version 1.10.x to 1.11.x, migrations not working properly. I'm using the "schema_filter" option and I realized that this option stops working.

I reviewed the changes and came to the conclusion that the following line should be added to the file "src/Kernel.php" (in "build" method):
```
$container->addCompilerPass(new DbalSchemaFilterPass());
```
After that, "schema_filter" option starts working, BUT only from one side (only DB tables are filtered, but entity classes not).
I looked into "migrations/lib/Doctrine/Migrations/Generator/DiffGenerator.php" file and I noticed that in method "createToSchema" (around line 104) is used deprecated method:
```
$this->dbalConfiguration->getFilterSchemaAssetsExpression()
```
In my opinion, should be this method (new one):
```
$this->dbalConfiguration->getSchemaAssetsFilter()
```
And "if" statment (in "foreach" loop) should be:
```
 if ($filterExpression($this->resolveTableName($tableName))) {
     continue;
}
```
When I'm change this two things, everything works perfectly.

#### Current behavior

Diff command not works properly with "schema_filter" option (only table names are filtered, entity classes not).
I have a entity classes with table name which should not be process while diff changes.

#### How to reproduce

1. Update "doctrine/doctrine-bundle" to latest version (1.11.0)
2. Enable "DbalSchemaFilterPass" in "src/Kernel.php" (in "build" method):
```
$container->addCompilerPass(new DbalSchemaFilterPass());
```
3. Set "schema_filter" option with some pattern (should be not include some entity clases)
4. Run "bin/console doctrine:migrations:diff"
5. Review new migration file. File should contains "CREATE" statements for entities which should be ignored.

#### Expected behavior

"schema_filter" option should be work properly (filtering table names and entity classes).

**Comment by bendavies** [comment]
https://github.com/doctrine/DoctrineBundle/pull/966

**Comment by mleko64** [comment]
Thanks for fast response.
Okey, this PR solving one problem - missing compiler pass registration.
What about second problem? In my opinion, method "createToSchema" in file "migrations/lib/Doctrine/Migrations/Generator/DiffGenerator.php" should be fixed.
Now, with new logic of "schema_filter" option, diff process not working properly.
Entity classes are not filtered correctly.

**Comment by bendavies** [comment]
hello @mleko64. I raised that PR before you raise this issue - I'm merely linking two related issues together. 
If you have a fix in mind to resolve the issue with `doctrine/migrations` feel fee to have a go at fixing it and raising a PR.

Thanks

**Comment by alcaeus** [comment]
This was fixed in 2.1.0, closing here.
